core/modules/filter/js/filter.filtered_html.admin.js

```diff
@@ -56,7 +56,27 @@ Backdrop.behaviors.filterFilterHtmlUpdating = {
 
   attach: function (context, settings) {
     var that = this;
-    $(context).find('[name="filters[filter_html][settings][allowed_html]"]').once('filter-filter_html-updating').each(function () {
+    // Respond to dialogs that are closed, updating the underlying form values.
+    $(".filter-dialog .ui-dialog-buttonpane .button-primary").click(function (e) {
```

Comment:
This might not be the best approach binding to the dialog button. If a validation error occurs within the dialog, the values might get updated with incorrect values. If the form is submitted without clicking on the button (e.g. submitting with the enter key) the values might not get saved either.

Instead of doing this client-side, I think it may be better to set the hidden element value in `filter_admin_format_configure_ajax()` as an ajax command, then all the code below can probably stay unmodified.